metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/lineage.py

```diff
@@ -752,20 +753,37 @@ def _create_lineage_map(
 
             # Try the sql parser first.
             if self.config.lineage_use_sql_parser:
+                logger.debug(
+                    f"Using sql parser for lineage extraction for destination table: {destination_table.table_identifier.get_table_name()}, queryType: {e.statementType}, query: {e.query}"
+                )
                 if e.statementType == "SELECT":
                     # We wrap select statements in a CTE to make them parseable as insert statement.
                     # This is a workaround for the sql parser to support the case where the user runs a query and inserts the result into a table..
-                    query = f"""create table `{destination_table.table_identifier.get_table_name()}` AS
-                    (
-                        {e.query}
-                    )"""
+                    try:
+                        parsed_queries = sqlglot.parse(e.query, "bigquery")
+                        if parsed_queries[-1]:
+                            query = f"""create table `{destination_table.get_sanitized_table_ref().table_identifier.get_table_name()}` AS
+                            (
+                                {parsed_queries[-1].sql(dialect='bigquery')}
+                            )"""
+                        else:
+                            query = e.query
+                    except Exception:
```

Comment:
now that we have the other PR merged, we should be able to revert this?

Reply:
This part is not full because we still need to wrap it into a create statement and get the last statement from the query.
The sanitized table name can be reverted but do you think it can cause any harm if we do here as well?